### PR TITLE
chore(grype): add sbom scan action

### LINKED ISSUE #2: adding SBOM scanning capability into the worklofw
Hello @marcofranssen, thanks for this great resource that you covered all the details about making software supply chains even more secure with the help of a bunch of open-source tools. I thought that maybe we can add SBOM scanning capability into the puzzle to scan all the packages available on SBOM results are in good shape, I'm proposing to use [scan-action](https://github.com/anchore/scan-action) from Anchore.

Please tell me what are you thinking about this?

Thanks

### PR DESCRIPTION
**Pull request by developer-guy**
Signed-off-by: Batuhan Apaydın <batuhan.apaydin@trendyol.com>

Fixes #2

### CODE REVIEW

File: .github/workflows/release-docker.yaml [hunk 1]
```diff
@@ -7,7 +7,7 @@ on:
 
 env:
   COSIGN_EXPERIMENTAL: 1
-  IMAGE_NAME: ghcr.io/marcofranssen/slsa-workflow-examples-docker
+  IMAGE_NAME: ghcr.io/${{ github.repository_owner }}/slsa-workflow-examples-docker
```
Comment (marcofranssen) [inline .github/workflows/release-docker.yaml:10]: nice trick to make it work on Forks.

File: .github/workflows/release-docker.yaml [hunk 2]
```diff
@@ -109,7 +109,11 @@ jobs:
 
       - name: Install Syft
         uses: anchore/sbom-action/download-syft@v0.7.0
-
+
```
Comment (marcofranssen) [inline .github/workflows/release-docker.yaml:112]:
```suggestion
```

File: .github/workflows/release-docker.yaml (hunk 1 above)
Comment (developer-guy) [reply .github/workflows/release-docker.yaml:10]: thanks 🙋🏻‍♂️

File: .github/workflows/release-docker.yaml (hunk 2 above)
Comment (developer-guy) [reply .github/workflows/release-docker.yaml:112]: removed

**Review by marcofranssen — APPROVED** [review]

**Pull request merged**